### PR TITLE
[drain-helper] remove usage of deprecated methods

### PR DESCRIPTION
**Pull request by tariq1890**

### CODE REVIEW

File: pkg/upgrade/drain_manager.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func (m *DrainManagerImpl) ScheduleNodesDrain(ctx context.Context, drainConfig *
 		GracePeriodSeconds:  -1,
 		Timeout:             time.Duration(drainSpec.TimeoutSecond) * time.Second,
 		PodSelector:         drainSpec.PodSelector,
-		OnPodDeletedOrEvicted: func(pod *corev1.Pod, usingEviction bool) {
+		OnPodDeletionOrEvictionFinished: func(pod *corev1.Pod, usingEviction bool, _ error) {
```
Comment (shivamerla) [inline pkg/upgrade/drain_manager.go:86]: Looks like the idea is to evaluate `error` field to determine if the eviction finished or failed. We should change the below logic accordingly. See e.g. [here](https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/drain/drain.go#L165).

Comment (tariq1890) [reply pkg/upgrade/drain_manager.go:86]: Agreed. Thanks for pointing this out!

Comment (tariq1890) [reply pkg/upgrade/drain_manager.go]: Please check now

File: pkg/upgrade/drain_manager.go [hunk 2]
```diff
@@ -83,12 +83,25 @@ func (m *DrainManagerImpl) ScheduleNodesDrain(ctx context.Context, drainConfig *
...
+					verbStr = "deletion failed"
+				} else {
+					verbStr = "deleted"
+				}
+			}
+			if err != nil {
+				m.log.V(consts.LogLevelWarning).Info("pod deletion/eviction failed", "error", err)
 			}
-			m.log.V(consts.LogLevelInfo).Info(fmt.Sprintf("%s pod from Node %s/%s", verbStr, pod.Namespace, pod.Name))
+			m.log.V(consts.LogLevelInfo).Info(fmt.Sprintf("pod %s/%s %s\n", pod.Namespace, pod.Name, verbStr))
```
Comment (tobiasgiese) [inline pkg/upgrade/drain_manager.go:104]: An alternative could be

```go
			log := m.log.WithValues("pod", pod.Name, "namespace", pod.Namespace, "eviction", usingEviction)
			if err != nil {
				log.V(consts.LogLevelWarning).Info("Drain Pod failed", "error", err)
				return
			}
			log.V(consts.LogLevelInfo).Info("Drain Pod finished")
```

Comment (tariq1890) [reply pkg/upgrade/drain_manager.go]: Done, thanks for the review @tobiasgiese ! Please check now

**Review by tobiasgiese — APPROVED** [review]

File: pkg/upgrade/drain_manager.go (hunk 2 above)
Comment (tobiasgiese) [reply pkg/upgrade/drain_manager.go]: Looks good to me now. Unfortunately I am not able to approve

**Review by adrianchiris — APPROVED** [review]

**Review by shivamerla — APPROVED** [review]
lgtm

**Pull request merged**